### DISCUSSION TITLE
Account for not-flat surface

### DISCUSSION THREAD

**Discussion by dostuffthatmatters** [tum-esm/stilt-at-esm#1] [Improvements]
The set of column receptors looks like the following schematic. Their position in space is given by the sun-position.

<img src="https://user-images.githubusercontent.com/29046316/142239978-630d74aa-9bbb-4ea8-a30c-9c94856d33c5.png" width="500px"/>

<br/><br/>

**Currently:** It is assumed that the ground is flat -> each receptors altitude above ground is assumed to be the altitude above ground at the instrument location

**Idea for improvement (by Friedrich):** The receptors height could be given as the actual altitude above the ground instead.

The following images shows the setup quite good:

<img src="https://user-images.githubusercontent.com/29046316/142242720-59bf5da6-6e15-4769-bc9d-5606e340246a.PNG" width="500px"/>

<br/><br/>

Each receptors altitude above ground is affecting the footprint modeling as well as the weighing of each receptor when computing one total column footprint. For Munich with a relatively flat terrain this is probably not differing a lot, but somewhere else it might be.

**Comment by FriedrichKlappenbach** [comment]
ToDos:
Obtain the Elevation data (Orography) for the used meteorological model (ERA5) by either:
a) use trajectories already calculated and extract "ZSFC" parameter and fill it into lat-lon grid
b) find the original file used by the modellers. (a 5 min. research did not yield a reasonable result)

re-write generate_receptors.r
- skip the use of map-files to calculate the scalingfactor (this can be obtained later on from the trajectories in a separate step)
- calculate the receptors using the surface elevation obtained in the first step.
